lib/parallel_cucumber/main.rb

```diff
@@ -70,18 +68,17 @@ def run
 
       @logger.info("Adding #{tests.count} tests to Queue")
       queue.enqueue(tests) unless tests.empty?
+      begin
+        Hooks.fire_before_workers(queue: queue)
+      rescue StandardError => e
+        trace = e.backtrace.join("\n\t")
+        @logger.warn("There was exception in before_workers hook #{e.message} \n #{trace}")
+      end
 
-      number_of_workers = determine_work_and_batch_size(count)
+      number_of_workers = determine_work_and_batch_size(queue.length)
```

Comment:
It seems you're missed tests from `directed_queue`.

Reply:
made a fix